src/pages/ServiceDetails/ServiceDetailsPage.tsx

```diff
@@ -161,7 +161,7 @@ class ServiceDetails extends React.Component<ServiceDetailsProps, ServiceDetails
   }
 
   componentDidMount() {
-    this.doRefresh();
+    this.promises.cancelAll();
```

Comment:
iirc there's several scenarios in `doRefresh` (traces, traffic, details ... ) .. did you check all scenarios are still ok on first load / page refresh?
(I can test that later, now just looking at the code)

Reply:
Oh, I think this is wrong, this PR shouldn't touch the ServiceDetailsPage.
Let me review, fix it.

Reply:
This is fixed now.